Add support for adding build and settings Gradle plugins

shanman190 opened the pull request.

knutwannheden commented:
@shanman190 Just for my understanding: Why do the new recipes and visitors declare an option named `pluginIdPattern` rather than just `pluginId`? AFAICT it ends up getting used by `FindPlugins`. Even though there the parameter of `FindPlugins#find()` is indeed also called `pluginIdPattern`, the corresponding option (`@Option` annotated field) of `FindPlugins` is called `pluginId` and there isn't any pattern matching taking place.

While I can see how a pattern could make sense for `FindPlugins`, I would in the case of `AddBuildPlugin` and `AddSettingPlugin` have expected the option to be called `pluginId`, since its purpose is to specify the ID of the plugin to add.

knutwannheden commented on rewrite-gradle/src/main/java/org/openrewrite/gradle/plugins/AddSettingPlugin.java:
```diff
@@ -0,0 +1,52 @@
+
+    @Override
+    public String getDescription() {
+        return "Add a Gradle settings plugin to `build.gradle(.kts)`.";
```
Should probably read `settings.gradle(.kts)`.

knutwannheden commented on rewrite-gradle/src/main/java/org/openrewrite/gradle/plugins/AddSettingPlugin.java:
```diff
@@ -0,0 +1,52 @@
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class AddSettingPlugin extends Recipe {
```
I am wondering if `AddSettingsPlugin` wouldn't be the more appropriate name.

shanman190 commented:
@knutwannheden, honestly, was copy and paste for the options initially from another recipe. I updated the description, but forgot to update the parameter name itself. 😅 I've made the updates that you've mentioned and fixed the test issue with PR #2599.

The visitor was actually existing, but I've updated it's parameter as it more aligns to the full pluginId and not a pattern like you suggested.

knutwannheden commented:
Excellent! I wasn't sure if I was missing something here.
